### PR TITLE
Update dependencies + remove deprecated code + fix

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Removing some deprecated code and fixing a potential undefined access issue. As a result I had to update a few things.

### CODE REVIEW

File: lib/rtc_monitor.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export class RTCMonitor extends EventEmitter {
         let totalJitter = 0;
         let totalLocalStats = 0;
         for (const [ssrc, stat] of Object.entries(localIn)) {
-            if (!this.stats.lastLocalIn[ssrc] || stat.timestamp <= this.stats.lastLocalIn[ssrc].timestamp) {
+            if (!this.stats.lastLocalIn[ssrc] || !this.stats.lastRemoteOut[ssrc] || stat.timestamp <= this.stats.lastLocalIn[ssrc].timestamp) {
```
Comment (streamer45) [inline lib/rtc_monitor.js:49]: This is the real fix. It can happen that `lastRemoteOut` is not available yet.

Comment (cpoile) [reply lib/rtc_monitor.js:49]: Did we encounter this happening? Or just preventative?

File: lib/rtc_peer.js [hunk 2]
```diff
@@ -264,36 +264,17 @@ export class RTCPeer extends EventEmitter {
...
-                        direction: 'sendonly',
-                        sendEncodings: this.config.simulcast && !isFirefox() ? DefaultSimulcastScreenEncodings : FallbackScreenEncodings,
-                        streams: [stream],
-                    });
-                    if ((opts === null || opts === void 0 ? void 0 : opts.codec) && trx.setCodecPreferences) {
-                        this.logger.logDebug('setting video codec preference', opts.codec);
-                        trx.setCodecPreferences([opts.codec]);
-                    }
-                    sender = trx.sender;
+                this.logger.logDebug('RTCPeer.addTrack: creating new transceiver on send');
```
Comment (cpoile) [inline lib/rtc_peer.js:267]: 👍

**Review by cpoile — APPROVED** [review]

File: lib/rtc_monitor.js (hunk 1 above)
Comment (streamer45) [reply lib/rtc_monitor.js:49]: @cpoile It happened this morning to @agarciamontoro

Comment (agarciamontoro) [reply lib/rtc_monitor.js:49]: Thanks for investigating this one!

**Pull request merged**